slack/client.go

```diff
@@ -0,0 +1,89 @@
+package slack
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/slack-go/slack"
+)
+
+// Client is a wrapper around the go-slack client
+type Client struct {
+	client   *slack.Client
+	channels Channels
+}
+
+// New returns a new Client if Slack notifications are enabled.
+// If not enabled, it returns a NoopClient
+func New(slackConfig *SlackConfig, apiToken string) (Clienter, error) {
+	if slackConfig == nil {
```

Comment:
Does this mean that to disable Slack we will still have to provide the enabled value in the slack config or this would fail?  It would be good if we still had the SLACK_ENABLED config value at a higher level to allow us to remove anything slack related if required in the future.